Implement tabbed navigation with Protocol, Stakeholders, and Proposals sections

JorikSchellekens opened the pull request:
This PR implements a tabbed navigation system for the Wei application with three main sections and enhanced proposal selection functionality.

## Changes Made

### Header Updates
- Updated tagline from 'Wei – Proposal Analyzer' to just 'Wei'

### New Tab System
- Created new Tabs component for navigation between sections
- Added three tabs: Protocol, Stakeholders, and Proposals
- Restructured main layout to accommodate tabbed interface

### Content Organization
- Moved existing proposal list and analysis functionality under Proposals tab
- Added placeholder content for Protocol and Stakeholders tabs
- Maintained all existing functionality within the Proposals tab

### Enhanced Search and Navigation Behavior
- Cmd-K space selection now only sets the space without changing tabs
- **NEW**: Proposal selection automatically switches to Proposals tab and selects the proposal
- **NEW**: When selecting a proposal via cmd+k search, the space/protocol automatically switches to match the proposal's space
- **NEW**: Selected proposal is automatically scrolled into view and highlighted in the left sidebar
- Preserved all existing search modal functionality

### Technical Implementation
- Added state management for active tab selection
- **NEW**: Enhanced handleSelectProposal to automatically set space from proposal data
- **NEW**: Added scrolling functionality to ProposalList component to ensure selected proposal visibility
- **NEW**: Added refs and useEffect hooks to handle smooth scrolling to selected proposals
- Maintained responsive design with proper grid layouts
- Preserved all existing styling and component structure

## User Experience Improvements
- **NEW**: Seamless navigation when selecting proposals from search - automatically switches context
- **NEW**: Visual feedback with selected proposal always visible and highlighted
- **NEW**: Intuitive workflow where proposal selection brings user to the right space and tab
- Tab navigation works as expected with proper state management
- Search behavior follows all specified requirements

## Testing
- All existing functionality preserved
- UI compiles without errors
- Tab navigation works as expected
- **NEW**: Proposal selection via search properly switches space and scrolls to proposal
- **NEW**: Selected proposals remain highlighted and visible in sidebar
- Search behavior follows all new requirements

JorikSchellekens committed:
Implement tabbed navigation with Protocol, Stakeholders, and Proposals sections

- Update header tagline from 'Wei – Proposal Analyzer' to 'Wei'
- Add new Tabs component for navigation between sections
- Restructure AnalyzerClient to use tab-based layout
- Move existing proposal list and analysis under Proposals tab
- Add placeholder content for Protocol and Stakeholders tabs
- Maintain search functionality without changing active tab
- Auto-switch to Proposals tab when selecting a proposal
- Preserve space selection behavior in search modal

JorikSchellekens committed:
Enhance proposal selection to auto-switch space and ensure visibility

- Modify handleSelectProposal to automatically set space/protocol from selected proposal
- Add scrolling functionality to ProposalList to ensure selected proposal is visible
- Selected proposal now scrolls into view when selected via search (cmd+k)
- Maintain highlight state of selected proposal in left sidebar
- Improve user experience when navigating proposals via search modal

JorikSchellekens committed:
feat: Enhance proposal selection with individual pages and improved navigation

- Create individual proposal pages with dynamic routes (/proposals/[id])
- Add linkable tabs and spaces with URL state management
- Implement conditional navigation behavior in proposal cards
- Show proposal list when no proposal is selected on main page
- Add back navigation with preserved filters
- Create collapsed dropdown for proposal content with title above
- Position back button under tabs for better UX
- Display analysis results below proposal content
- Maintain backward compatibility for main page analysis flow
- Add proper error handling and loading states for direct URL navigation
- Remove unused state variables and clean up imports

yehia67 commented on .cursor/rules/create-pr.mdc:
```diff
@@ -10,4 +10,7 @@ When asked to create a pr
 - make sure the ui compiles correctly with no errors
 - make sure all formatting checks are done correctly
 - make sure all changes are synced to origin
+- target unless otherwise specifid should be develop
```
```suggestion
- target unless otherwise specified should be develop
```

yehia67 commented on ui/src/components/proposals/proposal-page.tsx:
```diff
@@ -0,0 +1,442 @@
+    
+    const queryString = params.toString();
+    router.push(`/${queryString ? `?${queryString}` : ''}`);
+  };
```
```suggestion
  const buildUrlWithParams = (baseUrl: string, params: Record<string, string | null>) => {
  const searchParams = new URLSearchParams();
  Object.entries(params).forEach(([key, value]) => {
    if (value) searchParams.set(key, value);
  });
  const queryString = searchParams.toString();
  return `${baseUrl}${queryString ? `?${queryString}` : ''}`;
};

const handleBackClick = () => {
  const url = buildUrlWithParams('/', {
    space: selectedSpaceId,
    tab: activeTab
  });
  router.push(url);
};
```

yehia67 commented on ui/src/components/proposals/proposal-page.tsx:
```diff
@@ -0,0 +1,442 @@
+        analyzeProposal(proposal);
+      }
+    }
+  }, [allProposals, proposalId, selectedSpaceId, setSelectedSpaceId]);
```
```suggestion
  }, [analyzeProposal, allProposals, proposalId, selectedSpaceId, setSelectedSpaceId]);
```

JorikSchellekens committed:
Update .cursor/rules/create-pr.mdc

Co-authored-by: Yehia Tarek <yehia@nethermind.io>

JorikSchellekens committed:
Update ui/src/components/proposals/proposal-page.tsx

Co-authored-by: Yehia Tarek <yehia@nethermind.io>

JorikSchellekens committed:
Update ui/src/components/proposals/proposal-page.tsx

Co-authored-by: Yehia Tarek <yehia@nethermind.io>

JorikSchellekens committed:
Fix React hooks dependency issue in proposal-page.tsx

- Move analyzeProposal function declaration before useEffect to resolve temporal dead zone error
- Wrap analyzeProposal in useCallback to prevent unnecessary re-renders
- Add analyzeProposal to useEffect dependency array for proper React hooks compliance
- Resolves TypeScript compilation error and ESLint warning
